core/src/main/scala/kafka/tools/DumpLogSegments.scala

```diff
@@ -651,4 +651,17 @@ object DumpLogSegments {
     def checkArgs(): Unit = CommandLineUtils.checkRequiredArgs(parser, options, filesOpt)
 
   }
+
+  private[tools] def newDecoder(className: String, props: VerifiableProperties): Decoder[_] = {
+    val kclass = Class.forName(className).getConstructor(props.getClass).newInstance(props)
```

Comment:
This way will require all implementations of new decoder have to offer a constructor accepting properties. It seems to me that is not what we expected, as the props is always empty. Also, we should use Reconfigurable interface for consistency